truss/remote/baseten/api.py

```diff
@@ -202,72 +225,46 @@ def create_development_model_from_truss(
         resp = self._post_graphql_query(query_string)
         return resp["data"]["deploy_draft_truss"]
 
-    def deploy_chain(self, name: str, chainlet_data: List[b10_types.ChainletData]):
-        chainlet_data_strings = [
-            _chainlet_data_to_graphql_mutation(chainlet) for chainlet in chainlet_data
-        ]
-
-        chainlets_string = ", ".join(chainlet_data_strings)
-        query_string = f"""
-        mutation {{
-        deploy_chain(
-            name: "{name}",
-            chainlets: [{chainlets_string}]
-        ) {{
-            id
-            chain_id
-            chain_deployment_id
-        }}
-        }}
-        """
-        resp = self._post_graphql_query(query_string)
-        return resp["data"]["deploy_chain"]
-
-    def deploy_draft_chain(
-        self, name: str, chainlet_data: List[b10_types.ChainletData]
-    ):
-        chainlet_data_strings = [
-            _chainlet_data_to_graphql_mutation(chainlet) for chainlet in chainlet_data
-        ]
-        chainlets_string = ", ".join(chainlet_data_strings)
-        query_string = f"""
-        mutation {{
-        deploy_draft_chain(
-            name: "{name}",
-            chainlets: [{chainlets_string}]
-        ) {{
-            chain_id
-            chain_deployment_id
-        }}
-        }}
-        """
-        resp = self._post_graphql_query(query_string)
-        return resp["data"]["deploy_draft_chain"]
-
-    def deploy_chain_deployment(
+    def deploy_chain_atomic(
         self,
-        chain_id: str,
-        chainlet_data: List[b10_types.ChainletData],
+        entrypoint: b10_types.ChainletDataAtomic,
+        dependencies: List[b10_types.ChainletDataAtomic],
+        chain_id: Optional[str] = None,
+        chain_name: Optional[str] = None,
```

Comment:
Are there cases where `name` is `None`?

Reply:
Yes - when updating an existing Chain, only the Chain ID is passed in. When a new Chain is being created, only the Chain name is passed in.